docs/source/configuration_file.md

```diff
@@ -0,0 +1,139 @@
+#### debugging
+Bool. Allows extended verbosity and intermediate outputs.
+
+## Loader parameters
+#### bids_path
+String. Relative path of the BIDS folder.
+#### target_suffix
+Suffix list of the derivative file containing the ground-truth of interest (e.g. [`"_seg-manual"`, `"_lesion-manual"`]). If the list has a length greater than 1, then a multi-class model will be trained.
+#### contrasts
+- `train_validation`: list of image modalities loaded for the training and validation. If `multichannel`, then this list represent the different channels of the input tensors. Otherwise, the modalities are mixed and the model has only one input channel.
+- `test`: list of image modalities loaded in the testing dataset. Same comment than for `train_validation` regarding `multichannel`.
+- `balance`: Dict. Used to under-represent some modalities in the dataset: e.g. `{"T1w": 0.1}` will include only 10% of the available `T1w` images into the training/validation/test set. Please set `multichannel` to `false` if you are using this parameter.
+#### multichannel
+Bool. Indicated if more than a modality is used by the model. See details in both `train_validation` and `test` for the modalities that are input.
+#### slice_axis
+Choice between `"sagittal"`, `"coronal"`, and `"axial"`.
+This parameter decides the slice orientation on which the model will train.
+#### slice_filter
+Dict:
+- `filter_empty_input`: Bool. Filter empty images if `true`.
+- `filter_empty_mask`: Bool. Filter empty GT mask slices if `true`.
+#### roi
+Dict. of parameters about the region of interest
+- `suffix`: suffix of the derivative file containing the ROI used to crop (e.g. `"_seg-manual"`) with `ROICrop` as transform. Please use `null` if you do not want to use a ROI to crop.
+- `slice_filter_roi`: int. It filters ROI mask slices with less than this threshold of non zero voxels. Active when using `"ROICrop"` and inputing ROI file.
+
+## Split dataset
+#### fname_split
+Filename of a joblib file containing the list of training / validation / test subjects, used to ensure reproducible experiments
+#### random_seed
+Seed used by the random number generator to split the dataset between training/validation/testing.
+#### center_test
+List. List of centers to only include in the testing dataset.
+#### method
+Choice between `"per_patient"` (i.e. shuffle all subjects then splits) or `"per_center"` (split subject according to their acquisition centers).
+#### train_fraction
+Number between `0` and `1` representing the fraction of the dataset used as training set.
+#### test_fraction
+Number between `0` and `1` representing the fraction of the dataset used as test set. This parameter is only used if the `method` is `"per_patient"`.
+
+## Training parameters
+#### batch_size
+Integer.
+#### loss
+- `name`: Name of the loss function: Choice among the classes that are available [here](https://github.com/neuropoly/ivado-medical-imaging/blob/master/ivadomed/losses.py).
```

Comment:
i would define the hyperlink to the documentation file `losses.md`. I'm not 100% sure how to do the cross-ref in md files, so you'll need to figure out.

Reply:
Issue opened [here](https://github.com/neuropoly/ivado-medical-imaging/issues/281).